### ISSUE TITLE
Initializing with no mobiledoc dies after entering 2 characters

### ISSUE THREAD

**Issue by rlivsey** [bustlelabs/content-kit-editor#57]
I've initialized the editor without an initial `mobiledoc` as follows:

```javascript
let editor = new ContentKit.Editor(element, {});
```

All looks fine on the first keypress, but as soon as the second one is entered I get:

<img width="1346" alt="screenshot 2015-08-11 14 59 14" src="https://cloud.githubusercontent.com/assets/3740/9199303/a011c334-4039-11e5-8a51-e71bf65947f4.png">

The `element` here appears to be the `document` itself, so it looks like it's walking up out of the editor and not stopping until it hits the root.

It seems to be that `collectMarkups` is [being called](https://github.com/bustlelabs/content-kit-editor/blob/b4efbef396d84736b9465315e6f29f061875957c/src/js/parsers/post.js#L58-L70) with the `textNode` and `rootNode` being the same and then immediately calls `currentNode = textNode.parentNode` and so has now escaped the editor.

Should this check if `textNode === rootNode` first, or is the problem higher up?

**Comment by bantic** [comment]
@rlivsey Thanks for the report — the `element` in the `Editor` constructor should be something that you choose deliberately. Can you explain what you mean when you say it appears to be the `document`?

The `element` passed to the constructor should be a dom node directly, as well — the result of a `document.getElementById` call.

**Comment by rlivsey** [comment]
Here's the exact code I've got here, it's a shiny new Ember app using 1.13.5 using content-kit-editor which is npm linked into the project from a local build of master. Running on Safari & Chrome both have the same error.

```javascript
// app/components/content-editor/component.js
import Ember from 'ember';

export default Ember.Component.extend({
  didInsertElement() {
    const element = this.$(".editor")[0];
    this.editor = new ContentKit.Editor(element, {});
  },

  willDestroyElement() {
    this.editor.destroy();
  }
});
```

```handlebars
<!-- app/components/content-editor/template.hbs -->
<div class="editor"></div>
```

```handlebars
<!-- app/templates/application.hbs -->
{{content-editor}}
```

Running that and then entering two characters causes the error.

[`normalizeTagName`](https://github.com/bustlelabs/content-kit-editor/blob/c404d3b88871c6920690352392feb297521a9521/src/js/utils/dom-utils.js#L120) is called with `undefined`, because `document` doesn't have a tag name.

It appears to get `document` because `collectMarkups` in `reparseSection` ends up being called with the same `textNode` as `rootNode` and [immediately grabs the `parentNode` from `textNode`](https://github.com/bustlelabs/content-kit-editor/blob/c404d3b88871c6920690352392feb297521a9521/src/js/parsers/post.js#L60). 

```javascript
    // here textNode === rootNode 
    const  collectMarkups = (textNode, rootNode) =>{
      let markups = [];
      let currentNode = textNode.parentNode; 
      while (currentNode && currentNode !== rootNode) {  // this will never be true
        let markup = markupFromNode(currentNode);
        if (markup) {
          markups.push(markup);
        }

        currentNode = currentNode.parentNode;
      }
      return markups;
    };
```

As for why `textNode === rootNode` here, I'm not entirely sure yet - haven't had time to dig more & fully grok what's going on - so I'm not sure if that's the bug, or if it should be fine with them being the same thing.

Hope that helps!

**Comment by jacojoubert** [comment]
Here seems to be what is happening. When you initialize content-kit without a mobiledoc and start typing a default `p` node isn't created and you enter directly into the main editor node like this: 

![image](https://cloud.githubusercontent.com/assets/41799/9224122/0f1ac49e-40ce-11e5-939c-5fb2f760dfab.png)

Thus when you attempt to delete, press enter, or enter additional input it incorrectly identifies the editor as the `parentNode` and fails.

**Comment by mixonic** [comment]
Hey guys I have a branch with a fix for this. I fixed it while spiking out a fix for placeholders and our API for programmatic changes to the post. Will PR.

**Comment by mixonic** [comment]
Fixed in https://github.com/bustlelabs/content-kit-editor/pull/66, released in 0.3.2

**Comment by rlivsey** [comment]
Awesome, cheers
